autogpt_platform/backend/backend/api/features/integrations/router.py

```
@@ -371,6 +376,162 @@ async def callback(
+        )
+
+    device_code = valid_state.state_metadata.get("device_code")
+    if not device_code:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="State token is not for a device code flow",
+        )
+
+    try:
+        result = await handler.poll_for_tokens(device_code)
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)